### DISCUSSION TITLE
Add support for `using` Declarations and Explicit Resource Management

### DISCUSSION THREAD

**Discussion by GuestInCorle** [mobxjs/mobx#3840] [Ideas]
TypeScript 5.2 supports [`using` Declarations and Explicit Resource Management](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html).

Now it's possible to write this:
```TypeScript
{
  using _ = _action();
  obj.a = 1;
  obj.b = 2;
}
```

Instead of this:
```TypeScript
runInAction(() => {
  obj.a = 1;
  obj.b = 2;
});
```

**Comment by kubk** [comment]
A draft MR addressing it: https://github.com/mobxjs/mobx/pull/3841
